lib/stages/mergeBeamBuffer.hpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file
+ * @brief Merge the single frame beam buffer to a buffer with longer timespan.
+ *  - mergeBeamBuffer : public kotekan::Stage
+ */
+
+#ifndef MERGE_BEAM_BUFFER
+#define MERGE_BEAM_BUFFER
+
+#include "BeamMetadata.hpp" // for BeamMetadata
+#include "Config.hpp"       // for Config
+#include "Stage.hpp"
+#include "bufferContainer.hpp" // for bufferContainer
+
+#include <stdint.h> // for uint32_t
+#include <string>   // for string
+#include <vector>   // for vector
+
+
+using std::vector;
+
+
+/**
+ * @class mergeBeamBuffer
+ * @brief A stage to merge n single beam frame to one single
+ *        beam frame.
+ *
+ * @par Buffers
+ * @buffer in_buf Kotekan buffer of raw frames with BeamMetadata
+ *     followed by 4+4-bit voltage data.
+ *     @buffer_format Array of @c chars
+ *     @buffer_metadata BeamMetadata
+ * @buffer out_buf Kotekan buffer with frames containing multiple
+ *     block of metadata in the format of FreqIDMetadata and voltage data.
+ *     @buffer_format Array of @c uint32_t
+ *     @buffer_metadata MergedBeamMetadata
+ * @conf   samples_per_data_set  Int.    No. of samples.
+ * @conf   sub_frames_per_merged_frame  Int.   Number of frames to merge.
+ *
+ * @author Jing
+ *
+ *
+ */
+
+
```

Comment:
There are a lot of extra lines in this file, and some of the other files.   `clang-format` doesn't remove them, but generally it's generally nicer not to have more than one line between sections, and no blank lines between comments and the code they describe (since that can stop the auto docs from picking them correctly).
```suggestion
 * @author Jing
 *
 */
```